### ISSUE TITLE
Baggageclaim in Concourse 2.2.1 reaps volumes but containers live on

### ISSUE THREAD

**Issue by DanielJonesEB** [concourse/baggageclaim#6]
Symptom: Users get `insufficient subnets remaining in the pool` error message
Concourse version: 2.2.1
garden-runc: 0.9.0
Deployment method: BOSH
IaaS: AWS

```
# we're bumping along the limit of containers, despite only running a handful of pipelines
$ fly -t target workers
name                                  containers  platform  tags  team
0a04d3fd-14e2-47ae-9f80-1d05532f68ac  253         linux     none  none

$ date
Fri Oct 28 10:07:35 UTC 2016

# Find out how many containers are > 10 hours old
$ ps -ef --sort=start_time | grep "runc events" | grep Oct | wc -l
174

$ ps -ef --sort=start_time | grep "runc events"
 # pick a container that should have died already
root     21710  5544  0 Oct27 ?        00:00:12 runc events 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c

$ gaol shell 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c
error: hijack: Backend error: Exit status: 500, message: {"Type":"","Message":"unable to find user root: no matching entries in passwd file","Handle":""}

$ gaol properties 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c
concourse:resource-process  eae0f660-8d4c-4ec9-77eb-8d7776e53b68
concourse:volume-mounts {"7038a25d-ce4e-427e-5980-8e82913bf990":"/tmp/build/get"}
kawasaki.dns-servers
kawasaki.host-interface wvo2n9aol8ni-0
garden.network.container-ip 10.254.1.2
garden.network.external-ip  10.0.10.8
kawasaki.iptable-inst   vo2n9aol8ni
kawasaki.iptable-prefix w--
kawasaki.mtu    1500
concourse:resource-result   {
  "version": {
    "digest": "sha256:a1a112f981df3f101113f299b9d29826200e0ad3f0ae3330c9066b27fbbe1a2c"
  },
  "metadata": [
    {
      "name": "image",
      "value": "sha256:6cdfa"
    }
  ]
}

garden.grace-time   300000000000
kawasaki.bridge-interface   wbrdg-0afe0100
concourse:volumes   ["7038a25d-ce4e-427e-5980-8e82913bf990","612630cf-45b6-41d5-7abf-cb6515919e09"]
garden.network.host-ip  10.254.1.1
garden.state    created
kawasaki.container-interface    wvo2n9aol8ni-1
kawasaki.subnet 10.254.1.0/30
user

# Can't access either volume
$ ls /var/vcap/data/baggageclaim/volumes/live/7038a25d-ce4e-427e-5980-8e82913bf990
ls: cannot access /var/vcap/data/baggageclaim/volumes/live/7038a25d-ce4e-427e-5980-8e82913bf990: No such file or directory

$ ls /var/vcap/data/baggageclaim/volumes/live/612630cf-45b6-41d5-7abf-cb6515919e09
ls: cannot access /var/vcap/data/baggageclaim/volumes/live/612630cf-45b6-41d5-7abf-cb6515919e09: No such file or directory

# Garden thinks it should be managing this container
$ gaol list | grep 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c
1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c

# No result for container GUID in logs, but both volumes have been reaped by Baggage Claim
$ grep -e 612630cf-45b6-41d5-7abf-cb6515919e09 -e 7038a25d-ce4e-427e-5980-8e82913bf990 -e 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c baggageclaim.std*
baggageclaim.stdout.log:{"timestamp":"1477596534.502955675","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"612630cf-45b6-41d5-7abf-cb6515919e09","session":"69393","ttl":300}}
baggageclaim.stdout.log:{"timestamp":"1477600514.502540827","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"7038a25d-ce4e-427e-5980-8e82913bf990","session":"69791","ttl":300}}
```

If we then `bosh ssh` onto the ATC...

```
# There are logs going back a reasonable time
$ /var/vcap/sys/log/atc# ll
total 100212
drwxr-xr-x 2 vcap vcap     4096 Oct 28 01:00 ./
drwxr-x--- 9 root vcap     4096 Oct  6 20:02 ../
-rw-r--r-- 1 vcap vcap      186 Oct 27 09:58 atc.stderr.log
-rw-r--r-- 1 vcap vcap 34032368 Oct 28 10:39 atc.stdout.log
-rw-r--r-- 1 vcap vcap 52825246 Oct 28 01:00 atc.stdout.log.1
-rw-r--r-- 1 vcap vcap  2428611 Oct 27 09:15 atc.stdout.log.2.gz
-rw-r--r-- 1 vcap vcap  2528905 Oct 26 16:00 atc.stdout.log.3.gz
-rw-r--r-- 1 vcap vcap  2771483 Oct 25 23:15 atc.stdout.log.4.gz
-rw-r--r-- 1 vcap vcap  2819258 Oct 25 07:45 atc.stdout.log.5.gz
-rw-r--r-- 1 vcap vcap  2632093 Oct 24 16:30 atc.stdout.log.6.gz
-rw-r--r-- 1 vcap vcap  2548359 Oct 24 00:15 atc.stdout.log.7.gz

# no evidence of the container GUID being mentioned
$ zgrep 1de8a2b3-a72d-43ef-6bd2-e1a63a4b8c2c * | wc -l
0
```

**Comment by DanielJonesEB** [comment]
Interestingly, this seems like it might be driven by user behaviour.

When listing container processes by time, it seems a little odd that there is only a single orphan since midnight, but as soon as it gets to office hours we start getting one orphan every couple of minutes.

```
$ ps -ef --sort=start_time | grep "/proc/self/exe"
# SNIP!
root     30704     1  0 Oct27 ?        00:00:00 /proc/self/exe init
root      5486     1  0 Oct27 ?        00:00:00 /proc/self/exe init
root     21706     1  0 Oct27 ?        00:00:00 /proc/self/exe init
root     32505     1  0 Oct27 ?        00:00:00 /proc/self/exe init
root      1629     1  0 04:10 ?        00:00:00 /proc/self/exe init
root     20108     1  0 09:48 ?        00:00:00 /proc/self/exe init
root     20853     1  0 09:48 ?        00:00:00 /proc/self/exe init
root     22398     1  0 09:48 ?        00:00:00 /proc/self/exe init
root     31571     1  0 09:51 ?        00:00:00 /proc/self/exe init
root      1366     1  0 09:51 ?        00:00:00 /proc/self/exe init
root      8974     1  0 09:53 ?        00:00:00 /proc/self/exe init
root     10608     1  0 09:54 ?        00:00:00 /proc/self/exe init
root     13974     1  0 09:54 ?        00:00:00 /proc/self/exe init
root     24614     1  0 09:57 ?        00:00:00 /proc/self/exe init
root     27895     1  0 09:58 ?        00:00:00 /proc/self/exe init
root     30103     1  0 09:59 ?        00:00:00 /proc/self/exe init
root     30769     1  0 09:59 ?        00:00:00 /proc/self/exe init
```

**Comment by DanielJonesEB** [comment]
FWIW upgrading to 2.4.0 _seems_ to have fixed the problem. It's only been a couple of days, but we're not seeing container counts creep up.

We also added a second worker, and saw orphans only on the first worker, and not on the second.

**Comment by vito** [comment]
I'm gonna go ahead and close this; if it wasn't fixed in 2.4 it'll likely be fixed in concourse/concourse#629 at least.
